services/api-server/src/simcore_service_api_server/models/schemas/model_adapter.py

```diff
@@ -0,0 +1,123 @@
+# Models added here "cover" models from within the deployment in order to restore backwards compatibility
+
+from datetime import datetime
+from decimal import Decimal
+from typing import Annotated
+
+from models_library.api_schemas_api_server.pricing_plans import (
+    ServicePricingPlanGet as _ServicePricingPlanGet,
+)
+from models_library.api_schemas_webserver.product import (
+    GetCreditPrice as _GetCreditPrice,
+)
+from models_library.api_schemas_webserver.resource_usage import (
+    PricingUnitGet as _PricingUnitGet,
+)
+from models_library.api_schemas_webserver.wallets import (
+    WalletGetWithAvailableCredits as _WalletGetWithAvailableCredits,
+)
+from models_library.basic_types import IDStr, NonNegativeDecimal
+from models_library.resource_tracker import (
+    PricingPlanClassification,
+    PricingPlanId,
+    PricingUnitId,
+    UnitExtraInfo,
+)
+from models_library.users import GroupID
+from models_library.wallets import WalletID, WalletStatus
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    NonNegativeFloat,
+    NonNegativeInt,
+    PlainSerializer,
+)
+
+
+class GetCreditPriceLegacy(BaseModel):
+    model_config = ConfigDict(
+        populate_by_name=True,
+    )
+    product_name: str = Field(alias="productName")
+    usd_per_credit: (
+        Annotated[
+            NonNegativeDecimal,
+            PlainSerializer(float, return_type=NonNegativeFloat, when_used="json"),
+        ]
+        | None
+    ) = Field(
+        ...,
+        description="Price of a credit in USD. "
+        "If None, then this product's price is UNDEFINED",
+        alias="usdPerCredit",
+    )
+    min_payment_amount_usd: NonNegativeInt | None = Field(
```

Comment:
Please with new models let's start using the new annotations standard instead  of using `Field` in the "default location"

Reply:
Yes, I agree. But note https://github.com/ITISFoundation/osparc-simcore/pull/6866#discussion_r1871434990. I think the discrepancy you noticed between the schemas exported by fastapi and pydantic also holds here. I.e. using the `Field` in the `Annotation` with an `alias` doesn't seem to go into the schema correctly.